plugins/talk-plugin-moderation-actions/client/containers/BanUserAction.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import {compose} from 'react-apollo';
+import {openDialog} from '../actions';
+import {bindActionCreators} from 'redux';
+import BanUserAction from '../components/BanUserAction';
+import {connect, excludeIf} from 'plugin-api/beta/client/hocs';
+
+class BanUserActionContainer extends React.Component {
+  onBanUser = () => {
+    this.props.openDialog(this.props.comment);
+  }
+
+  render() {
+    return <BanUserAction 
+      onBanUser={this.onBanUser}
+      comment={this.props.comment}
+      openDialog={this.props.openDialog}
+    />;
+  }
+}
+
+const mapDispatchToProps = (dispatch) =>
+  bindActionCreators({
+    openDialog
+  }, dispatch);
+
+const enhance = compose(
+  connect(null, mapDispatchToProps),
+  excludeIf(({root: {me}, comment}) => !me || me.id === comment.user.id),
```

Comment:
This only works when this component is being used inside a slot, which is not the case.

Reply:
Oh, It makes sense. Is there something else I can use to have the same behavior? Just to stop the execution of the compose... :thinking:

Reply:
Just go back to the parent where you include this component, and add the logic there not to include it in render when it's the same user.

Reply:
Yeah. I was wondering if there was something in the compose that we could add.

Reply:
Will do!